rubin_sim/maf/metrics/tgaps.py

```diff
@@ -255,3 +256,48 @@ def run(self, dataSlice, slicePoint=None):
         else:
             result = self.badval
         return result
+
+
+class NightTimespanMetric(BaseMetric):
+    """Calculate the maximum time span covered in each night, report the `percentile` value of all timespans.
+
+    Parameters
+    ----------
+    percentile : `float`, opt
+        Percentile value to report. Default 75th percentile.
+    nightCol : `str`, opt
+        Name of the night column. Default 'night'.
+    mjdCol : `str`, opt
+        Name of the MJD visit column. Default 'observationStartMJD'.
+    """
+
+    def __init__(
+        self, percentile=75, nightCol="night", mjdCol="observationStartMJD", **kwargs
+    ):
+        self.percentile = percentile
+        self.nightCol = nightCol
+        self.mjdCol = mjdCol
+        if "metricName" in kwargs:
+            metricName = kwargs["metricName"]
+            del kwargs["metricName"]
+        else:
+            metricName = f"{percentile}th Percentile Intranight Timespan"
+        super().__init__(
+            col=[self.nightCol, self.mjdCol],
+            units="minutes",
+            metricName=metricName,
+            **kwargs,
+        )
+
+    def run(self, dataSlice, slicePoint=None):
+        data = np.sort(dataSlice, order=self.mjdCol)
+        unights, counts = np.unique(data[self.nightCol], return_counts=True)
+        unights = unights[np.where(counts > 1)]
+        if len(unights) == 0:
+            result = self.badval
+        else:
+            nstart = np.searchsorted(data[self.nightCol], unights, side="left")
+            nend = np.searchsorted(data[self.nightCol], unights, side="right") - 1
+            tspans = (data[self.mjdCol][nend] - data[self.mjdCol][nstart]) * 24.0 * 60.0
```

Comment:
Do you want to include cases where tspans=0? I'm not sure if it would be more informative clipping off instances where there wasn't a second observation or not.

Reply:
This is a good question. I thought briefly about it when creating the metric and decided that it was better to look at only cases where there are at least two visits .. it's not a super obvious answer though. You could definitely say that actually looking at any night where there was any visit was reasonable, and then using the total of all nights with visits as the basis for generating the X percentile to return. I don't think there is a clear answer, so for now, I'll leave it like this.

Reply:
I think the way the metric is written it is including nights with only one visit. The `unights` variable gets clipped, but `data` doesn't get the low count nights removed, so `tspans` is still going to have lots of zeros in it.

Reply:
No wait, it's doing it the right way. I always forget the order on `searchsorted`.

Reply:
But also - the `unights` are the nights where the count is >1, not just >0.